builder/ecs/step_config_eip.go

```diff
@@ -78,16 +78,34 @@ func (s *stepConfigAlicloudEIP) Run(ctx context.Context, state multistep.StateBa
 	}
 
 	if s.SSHPrivateIp {
-		ipaddresses := instance.VpcAttributes.PrivateIpAddress.IpAddress
-		if len(ipaddresses) == 0 {
+		_, err := client.WaitForExpected(&WaitForExpectArgs{
+			RequestFunc: func() (responses.AcsResponse, error) {
+				describeInstancesRequest := ecs.CreateDescribeInstancesRequest()
+				describeInstancesRequest.InstanceIds = fmt.Sprintf("[\"%s\"]", instance.InstanceId)
+				return client.DescribeInstances(describeInstancesRequest)
+			},
+			EvalFunc: func(response responses.AcsResponse, err error) WaitForExpectEvalResult {
+				if err != nil {
+					return WaitForExpectToRetry
+				}
+				instancesResponse := response.(*ecs.DescribeInstancesResponse)
+				instance = &instancesResponse.Instances.Instance[0]
+				ipAddresses := instance.VpcAttributes.PrivateIpAddress.IpAddress
+				if len(ipAddresses) == 0 {
+					return WaitForExpectToRetry
+				}
+				ipaddress = ipAddresses[0]
+				state.Put("ipaddress", ipaddress)
```

Comment:
I think this should stay in its previous position, otherwise only the private IP will be set in state, and not the public one if that's what is chosen.

Reply:
Thanks for the comment, you are right.